tools/qmlformat.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+"""
+Small qmlformat wrapper that warns the user if the tool is not installed.
+"""
+import argparse
+import shutil
+import subprocess
+import pathlib
+import sys
+
+QMLFORMAT_MISSING_MESSAGE = """
+qmlformat is not installed. It is included in Qt 5.15 and later. If that Qt
+version is not available on your system, please use the SKIP environment
+variable when committing:
+
+    $ SKIP=qmlformat git commit
+"""
+
+
+def main(argv=None):
+    qmlformat_executable = shutil.which("qmlformat")
+    if not qmlformat_executable:
+        print(QMLFORMAT_MISSING_MESSAGE.strip(), file=sys.stderr)
+        return 1
```

Comment:
Here we return exit code 1 so that the user is prompted to either install qmlformat or skip the hook manually.

We could also return exit code 0. Thdi would be even more convenient, but then the hook shows up as "passed" and the user has no way of knowing if qmlformat actually ran or not. This is an issue, especially on CI.

Reply:
I can imagine that it can leads to extra hassle when resolving conflicts, because it does not run on diffs only. 

So I would prefer exit code 0 version, since it will be visible in our CI anyway. 
That follows the approach that the whole pre-commit is optional anyway. Compared to that. this is one only a side issue, where we should not annoy a contributor that has just decided to install pre-commit.

Reply:
> So I would prefer exit code 0 version, since it will be visible in our CI anyway.

We cannot be sure about that. It will display as `passed` on CI in any case, even if your CI image does not actually have qmlformat.

I think the extra hassle is manageable. Just run this once:

    $ echo SKIP=qmlformat >> ~/.profile

And don't forget remove it after upgrading to Qt 5.15 ;-).

Reply:
Yes ok. Ready to merge IMHO.

Just an idea, considering that we should not forget this .profile value.
How about use: 
qmake --version
To exit 0 if qmlformat cannot be installed.